### PR TITLE
Provider agreement events

### PR DESCRIPTION
**Pull request by mgwilliam**

### CODE REVIEW

File: src/SFA.DAS.Events.Application/Queries/GetAgreementEvents/GetAgreementEventsRequestValidator.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+namespace SFA.DAS.Events.Application.Queries.GetAgreementEvents
+{
+    public class GetAgreementEventsRequestValidator : AbstractValidator<GetAgreementEventsRequest>
+    {
+        public GetAgreementEventsRequestValidator()
+        {
+            RuleFor(from => from.FromDateTime).LessThanOrEqualTo(to => to.ToDateTime);
+            RuleFor(request => request.PageNumber).GreaterThanOrEqualTo(1);
+            RuleFor(request => request.PageSize).InclusiveBetween(1, 10000); //todo: determine appropriate max page size
```
Comment (leewadhams) [inline src/SFA.DAS.Events.Application/Queries/GetAgreementEvents/GetAgreementEventsRequestValidator.cs]: Has this been determined?

File: src/SFA.DAS.Events.Application/Queries/GetAgreementEvents/GetAgreementEventsQueryHandler.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        public async Task<GetAgreementEventsResponse> Handle(GetAgreementEventsRequest request)
+        {
+            Validate(request);
+
+            var events = await _agreementEventRepository.GetByDateRange(request.FromDateTime, request.ToDateTime, request.PageSize, request.PageNumber, request.FromEventId);
+
+            return new GetAgreementEventsResponse { Data = MapFrom(events)};
+        }
+
+        private static IEnumerable<AgreementEvent> MapFrom(IEnumerable<AgreementEvent> source)
```
Comment (leewadhams) [inline src/SFA.DAS.Events.Application/Queries/GetAgreementEvents/GetAgreementEventsQueryHandler.cs]: What is this method for?

File: src/SFA.DAS.Events.Application/Queries/GetApprenticeshipEvents/GetApprenticeshipEventsQueryHandler.cs [hunk 3]
```diff
@@ -28,10 +28,10 @@ public GetApprenticeshipEventsQueryHandler(IApprenticeshipEventRepository appren
 
         private static IEnumerable<ApprenticeshipEvent> MapFrom(IEnumerable<ApprenticeshipEvent> source)
```
Comment (leewadhams) [inline src/SFA.DAS.Events.Application/Queries/GetApprenticeshipEvents/GetApprenticeshipEventsQueryHandler.cs]: What is this method for?

Comment (mgwilliam) [inline src/SFA.DAS.Events.Application/Queries/GetApprenticeshipEvents/GetApprenticeshipEventsQueryHandler.cs]: for mapping entities to api types (there's a separate library for api client types)

File: src/SFA.DAS.Events.Application/Queries/GetAgreementEvents/GetAgreementEventsQueryHandler.cs (hunk 2 above)
Comment (mgwilliam) [inline src/SFA.DAS.Events.Application/Queries/GetAgreementEvents/GetAgreementEventsQueryHandler.cs]: for mapping entities to api types (there's a separate library for api client types)

File: src/SFA.DAS.Events.Application/Queries/GetAgreementEvents/GetAgreementEventsRequestValidator.cs (hunk 1 above)
Comment (mgwilliam) [inline src/SFA.DAS.Events.Application/Queries/GetAgreementEvents/GetAgreementEventsRequestValidator.cs]: the 10,000 max is a guess for now (hence leaving the todo comment)

**Pull request merged**